corehq/apps/users/models.py

```diff
@@ -1214,6 +1241,29 @@ def user_session_data(self):
             '{}_last_name'.format(SYSTEM_PREFIX): self.last_name,
             '{}_phone_number'.format(SYSTEM_PREFIX): self.phone_number,
         })
+
+        domain = None
+        try:
+            domain = self.domain
+        except AttributeError:
+            domains = self.get_domains()
+            if domains:
+                domain = domains[0]
```

Comment:
Yeah, I didn't want to change behaviour here which is why I did this. I'm not sure if you can really use user_data on a web user...

Reply:
I dug into this question as part of the user profiles work. It looks like web users do not support user data, but they did support it in the past, at least for some certain projects, which is why `user_data` is an attribute of `CouchUser` and not `CommCareUser`. I want to say ews-ghana had some kind of extended web user that could have user data. I looked at constraining user data to mobile users and ended up deciding it would increase my scope too much.